RA failure w/ seed=2168839148116956066

```
---- STDOUT -----------------------------------------
Making a frame with nrows=97, ncols=3
  types: bool=0, int=1, float=1, str=1
  missing values: 0.938
Launching an attack for 52 rounds
[02] Applying row slice (12, 92, 1) -> nrows = 80
[03] Applying column slice (None, 2, 1) -> ncols = 2
[01] Setting nrows to 174
[05] Rbinding frame with itself 4 times -> nrows = 870
[04] Cbinding frame with itself 6 times -> ncols = 14
[05] Rbinding frame with itself 2 times -> nrows = 2610
[02] Applying row slice (318, 2404, 2) -> nrows = 1043
[02] Applying row slice (207, 1020, 3) -> nrows = 271
[02] Applying row slice (None, None, 1) -> nrows = 271
[05] Rbinding frame with itself 1 times -> nrows = 542
[02] Applying row slice (None, 491, 2) -> nrows = 246
[02] Applying row slice (25, 233, 2) -> nrows = 104
[04] Cbinding frame with itself 4 times -> ncols = 70
[01] Setting nrows to 62
[02] Applying row slice (None, None, -1) -> nrows = 62
[02] Applying row slice (None, 9, -2) -> nrows = 26
[01] Setting nrows to 23
[04] Cbinding frame with itself 1 times -> ncols = 140
[05] Rbinding frame with itself 4 times -> nrows = 115
[05] Rbinding frame with itself 3 times -> nrows = 460
[05] Rbinding frame with itself 2 times -> nrows = 1380
[02] Applying row slice (None, 152, -1) -> nrows = 1227
[02] Applying row slice (None, None, 1) -> nrows = 1227
[02] Applying row slice (190, None, 1) -> nrows = 1037
[02] Applying row slice (None, None, 2) -> nrows = 519
[01] Setting nrows to 3405
[02] Applying row slice (None, 3159, 5) -> nrows = 632
[01] Setting nrows to 806
[04] Cbinding frame with itself 1 times -> ncols = 280
[02] Applying row slice (None, None, 5) -> nrows = 162
[04] Cbinding frame with itself 1 times -> ncols = 560
[02] Applying row slice (151, None, -2) -> nrows = 76
[04] Cbinding frame with itself 1 times -> ncols = 1120
[02] Applying row slice (70, None, -1) -> nrows = 71
[05] Rbinding frame with itself 2 times -> nrows = 213

---- STDERR -----------------------------------------
Traceback (most recent call last):
  File "random_attack.py", line 400, in <module>
    ra.attack()
  File "random_attack.py", line 36, in attack
    self.attack_frame(frame)
  File "random_attack.py", line 46, in attack_frame
    action(self, frame)
  File "random_attack.py", line 86, in rbind_self
    frame.rbind([frame] * t)
  File "random_attack.py", line 351, in rbind
    self.df.rbind(*[iframe.df for iframe in frames])
  File "/Users/pasha/py36/lib/python3.6/site-packages/typesentry/config.py", line 155, in fdecorated
    ret = f(*args, **kws)
  File "/Users/pasha/py36/lib/python3.6/site-packages/datatable/dt_append.py", line 143, in _rbind
    _dt.rbind(len(final_names), spec)
AssertionError: Assertion 'mbuf.is_writable() && strbuf.is_writable()' failed in "c/column_string.cc", line 401
```

MRE:
```
>>> import datatable as dt
>>> f0 = dt.Frame([None, None, None, "foo"])
>>> f0.nrows = 2
>>> f0.rbind(f0)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/pasha/py36/lib/python3.6/site-packages/typesentry/config.py", line 155, in fdecorated
    ret = f(*args, **kws)
  File "/Users/pasha/github/datatable/datatable/dt_append.py", line 143, in _rbind
    _dt.rbind(len(final_names), spec)
AssertionError: Assertion 'mbuf.is_writable() && strbuf.is_writable()' failed in "c/column_string.cc", line 401
```

--- a/c/column_string.cc
+++ b/c/column_string.cc
@@ -398,7 +398,8 @@ void StringColumn<T>::rbind_impl(std::vector<const Column*>& columns,
   // Reallocate the column
   mbuf.resize(new_mbuf_size);
   strbuf.resize(new_strbuf_size);
-  xassert(mbuf.is_writable() && strbuf.is_writable());
+  xassert(mbuf.is_writable() || new_mbuf_size == 0);
+  xassert(strbuf.is_writable() || new_strbuf_size == 0);
   nrows = new_nrows;
   T* offs = offsets_w();
 
@@ -428,10 +429,13 @@ void StringColumn<T>::rbind_impl(std::vector<const Column*>& columns,
         T off = col_offsets[j];
         *offs++ = off + curr_offset;
       }
-      const MemoryRange& col_strbuf = static_cast<const StringColumn<T>*>(col)->strbuf;
-      void* target = strbuf.wptr(static_cast<size_t>(curr_offset));
-      std::memcpy(target, col_strbuf.rptr(), col_strbuf.size());
-      curr_offset += col_strbuf.size();
+      auto strcol = static_cast<const StringColumn<T>*>(col);
+      size_t sz = strcol->strbuf.size();
+      if (sz) {
+        void* target = strbuf.wptr(static_cast<size_t>(curr_offset));
+        std::memcpy(target, strcol->strbuf.rptr(), sz);
+        curr_offset += sz;
+      }
     }
     delete col;
   }

--- a/tests/munging/test_dt_append.py
+++ b/tests/munging/test_dt_append.py
@@ -363,3 +363,12 @@ def test_rbind_modulefn():
     f3 = dt.rbind(f0, f1)
     f3.internal.check()
     assert f3.topython()[0] == f0.topython()[0] + f1.topython()[0]
+
+
+def test_issue1292():
+    f0 = dt.Frame([None, None, None, "foo"])
+    f0.nrows = 2
+    f0.rbind(f0)
+    f0.internal.check()
+    assert f0.topython() == [[None] * 4]
+    assert f0.stypes == (stype.str32,)

